mls-rs-uniffi/src/lib.rs

```diff
@@ -0,0 +1,528 @@
+        Self { _inner: inner }
+    }
+}
+
+/// Light-weight wrapper around a [`mls_rs::group::ReceivedMessage`].
+#[derive(Clone, Debug, uniffi::Enum)]
+pub enum ReceivedMessage {
+    /// A decrypted application message.
+    ApplicationMessage {
+        sender: Arc<SigningIdentity>,
+        data: Vec<u8>,
+    },
+
+    /// A new commit was processed creating a new group state.
+    Commit { committer: Arc<SigningIdentity> },
+
+    /// A proposal was received.
+    Proposal {
+        sender: Arc<SigningIdentity>,
+        proposal: Arc<Proposal>,
+    },
+
+    /// Validated GroupInfo object.
+    GroupInfo,
+    /// Validated welcome message.
+    Welcome,
+
+    /// Validated key package.
+    KeyPackage { key_package: Arc<KeyPackage> },
+}
+
+/// Supported cipher suites.
+///
+/// This is a subset of the cipher suites found in
+/// [`mls_rs::CipherSuite`].
+#[derive(Copy, Clone, Debug, uniffi::Enum)]
+pub enum CipherSuite {
+    // TODO(mgeisler): add more cipher suites.
+    Curve25519Aes128,
+}
+
+impl From<CipherSuite> for mls_rs::CipherSuite {
+    fn from(cipher_suite: CipherSuite) -> mls_rs::CipherSuite {
+        match cipher_suite {
+            CipherSuite::Curve25519Aes128 => mls_rs::CipherSuite::CURVE25519_AES128,
+        }
+    }
+}
+
+/// Generate a MLS signature keypair.
+///
+/// This will use the default mls-lite crypto provider.
+///
+/// See [`mls_rs::CipherSuiteProvider::signature_key_generate`]
+/// for details.
+#[cfg_attr(not(mls_build_async), maybe_async::must_be_sync)]
+#[uniffi::export]
+pub async fn generate_signature_keypair(
+    cipher_suite: CipherSuite,
+) -> Result<SignatureKeypair, Error> {
+    let crypto_provider = mls_rs_crypto_openssl::OpensslCryptoProvider::default();
+    let cipher_suite_provider = crypto_provider
+        .cipher_suite_provider(cipher_suite.into())
+        .ok_or(MlsError::UnsupportedCipherSuite(cipher_suite.into()))?;
+
+    let (secret_key, public_key) = cipher_suite_provider
+        .signature_key_generate()
+        .await
+        .map_err(|err| MlsError::CryptoProviderError(err.into_any_error()))?;
+
+    Ok(SignatureKeypair {
+        cipher_suite,
+        public_key: Arc::new(public_key.into()),
+        secret_key: Arc::new(secret_key.into()),
+    })
+}
+
+/// An MLS client used to create key packages and manage groups.
+///
+/// See [`mls_rs::Client`] for details.
+#[derive(Clone, Debug, uniffi::Object)]
+pub struct Client {
+    inner: mls_rs::client::Client<Config>,
+}
+
+#[cfg_attr(not(mls_build_async), maybe_async::must_be_sync)]
+#[uniffi::export]
+impl Client {
+    /// Create a new client.
+    ///
+    /// The user is identified by `id`, which will be used to create a
+    /// basic credential together with the signature keypair.
+    ///
+    /// See [`mls_rs::Client::builder`] for details.
+    #[uniffi::constructor]
+    pub fn new(id: Vec<u8>, signature_keypair: SignatureKeypair) -> Self {
```

Comment:
Lets make an issue to make this a SigningIdentity and expose support for x509 as well. We can probably do that after the provider traits are exposed

Reply:
Sounds good, I created #85 for this.